### PR TITLE
Renesas: general renesas cleanups

### PR DESCRIPTION
**Pull request by perigoso**
A couple generic cleanups, initially motivated by going through Renesas documentation.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [ ] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

**Review by dragonmux — CHANGES_REQUESTED** [review]
Only spotted one potential misstep in this PR, otherwise it all looks good. Once the review point is addressed, we're happy to merge this.

Thanks for the contribution Perigoso!

File: src/target/renesas.c [hunk 1]
```diff
@@ -291,31 +295,36 @@ typedef struct renesas_priv {
...
 		pnrr[i] = target_mem_read32(t, base + i * 4U);
 
+	/* Write bytewise into provided container */
 	if (base == RENESAS_FIXED1_PNR) {
 		/* Renesas... look what you made me do...  */
 		/* reverse order, see 'Part numbering scheme' note for context */
 		for (size_t i = 0U; i < 13U; i++)
 			pnr[i] = pnrr[3U - (i + 3U) / 4U] >> (24U - ((i + 3U) & 3U) * 8U); /* & 3U == % 4U */
-		memset(pnr + 13U, 0x20, 3);
+		memset(pnr + 13U, '\x20', 3);
```
[the remark this answers is not in the document]
Comment (perigoso) [reply src/target/renesas.c:327]: This change was less about correctness, and more about intent, shows we are populating the 3 last bytes with the space character and not with a 0x20 hex which where it comes from is not immediately clear
that said this isn't much of an improvement (maybe worse lol?), maybe I should use the actual space char instead `' '`

[the remark this answers is not in the document]
Comment (dragonmux) [reply src/target/renesas.c:327]: Ah, yes, that's a fair call - in that case, using an actual space character literal makes considerably more sense in context.

[the remark this answers is not in the document]
Comment (perigoso) [reply src/target/renesas.c]: Opted to just add an extra comment for context

**Review by dragonmux — APPROVED** [review]
LGTM, as this is mostly `const`-correctness and comments cleanup we'll merge this and anyone using the Renesas support who finds it breaks because of this can open an issue ticket.

**Pull request merged**